### PR TITLE
Added tsdownsample as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/staged-recipes @conda-forge/rust Need some help here, looks like I'm able to get this building via Rust/Maturin for Linux and OSX but I'm not able to figure out what's wrong with the Windows build. Maybe it just doesn't work on Windows? But I'm not sure

File: recipes/tsdownsample/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+channel_sources:
+  - conda-forge/label/rust_dev,conda-forge
```
Comment (ocefpaf) [inline recipes/tsdownsample/conda_build_config.yaml:2]: Are you using any dev pkgs below? If not let's remove this.

**Comment by ocefpaf** [comment]
> Maybe it just doesn't work on Windows? But I'm not sure

It probably does but Windows is always a bit more difficult. If you don't care about the Windows builds you can skip it here and open an issue in the feedstock so folks with more knowledge of Windows can tackle that later.

File: recipes/tsdownsample/conda_build_config.yaml (hunk 1 above)
Comment (thewchan) [reply recipes/tsdownsample/conda_build_config.yaml:2]: Yeah this doesn't work without using the nightly

**Comment by thewchan** [comment]
@ocefpaf okay let's at least get osx and linux working first I guess.

**Pull request merged**